server/apps/publish/formatters/anpa_formatter.py

```diff
@@ -18,94 +18,91 @@
+                anpa.append(article.get('priority', 'r').encode('ascii'))
+                anpa.append(b'\x20')
+
+                anpa.append(category['qcode'].encode('ascii'))
+
+                anpa.append(b'\x13')
+                # format identifier
+                if article['type'] == 'preformatted':
+                    anpa.append(b'\x12')
+                else:
+                    anpa.append(b'\x11')
+                anpa.append(b'\x20')
+
+                # keyword
+                keyword = 'bc-{}'.format(article.get('slugline', '')).replace(' ', '-')
+                keyword = keyword[:24] if len(keyword) > 24 else keyword
+                anpa.append(keyword.encode('ascii'))
+                anpa.append(b'\x20')
+
+                # version field
+                anpa.append(b'\x20')
+
+                # reference field
+                anpa.append(b'\x20')
+
+                # filing date
+                anpa.append('{}-{}'.format(article['_updated'].strftime('%m'),
+                                           article['_updated'].strftime('%d')).encode('ascii'))
+                anpa.append(b'\x20')
+
+                # add the word count
+                anpa.append(str(article.get('word_count', '0000')).zfill(4).encode('ascii'))
+                anpa.append(b'\x0D\x0A')
+
+                anpa.append(b'\x02')  # STX
+
+                # Add the actual story bits
+                anpa.append(article.get('headline', '').encode('ascii', 'replace'))
+                anpa.append(b'\x0D\x0A')
+
+                keyword = article.get('slugline', '').encode('ascii', 'ignore')
+                anpa.append(keyword)
+                take_key = article.get('anpa_take_key', '').encode('ascii', 'ignore')
+                anpa.append((b'\x20' + take_key) if len(take_key) > 0 else b'')
+                anpa.append(b'\x0D\x0A')
+
+                if article['type'] == 'preformatted':
+                    anpa.append(article.get('body_html', '').encode('ascii', 'replace'))
+                else:
+                    anpa.append(BeautifulSoup(article.get('body_html', '')).text.encode('ascii', 'replace'))
+
+                anpa.append(b'\x0D\x0A')
+                if article.get('more_coming', False):
+                    anpa.append('MORE'.encode('ascii'))
+                else:
+                    anpa.append(article.get('source', '').encode('ascii'))
+                sign_off = article.get('sign_off', '').encode('ascii')
+                anpa.append((b'\x20' + sign_off) if len(sign_off) > 0 else b'')
+                anpa.append(b'\x0D\x0A')
+
+                anpa.append(b'\x03')  # ETX
+
+                # time and date
+                anpa.append(datetime.datetime.now().strftime('%d-%m-%y %H-%M-%S').encode('ascii'))
+
+                anpa.append(b'\x04')  # EOT
+                anpa.append(b'\x0D\x0A\x0D\x0A\x0D\x0A\x0D\x0A\x0D\x0A\x0D\x0A\x0D\x0A\x0D\x0A')
+
+                docs.append((pub_seq_num, b''.join(anpa)))
```

Comment:
Do we need to increment the pub_seq_number as every one of the formatted_item will be a delivery? The worst case would be the same subscriber getting the same story with different anpa-category but same sequence.